lib/auth/auth.go

```diff
@@ -4378,6 +4384,41 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
 	return req, nil
 }
 
+// updateAccessRequestWithAdditionalReviewers will update the given access request with additional reviewers given the promotions
+// created for the access request.
+func updateAccessRequestWithAdditionalReviewers(ctx context.Context, req types.AccessRequest, accessLists services.AccessListsGetter, promotions *types.AccessRequestAllowedPromotions) {
+	if promotions == nil {
+		return
+	}
+
+	// For promotions, add in access list owners as additional suggested reviewers
+	var additionalReviewers []string
+
+	// Iterate through the promotions, adding the owners of the corresponding access lists as reviewers.
+	for _, promotion := range promotions.Promotions {
+		accessList, err := accessLists.GetAccessList(ctx, promotion.AccessListName)
+		if err != nil {
+			log.WithError(err).Warn("Failed to get access list, skipping additional reviewers")
+			break
+		}
+
+		for _, owner := range accessList.GetOwners() {
+			additionalReviewers = append(additionalReviewers, owner.Name)
+		}
+	}
+
+	// Only modify the original request if additional reviewers were found.
+	if len(additionalReviewers) > 0 {
+		originalNumSuggestedReviewers := len(req.GetSuggestedReviewers())
+		suggestedReviewers := make([]string, originalNumSuggestedReviewers+len(additionalReviewers))
+		copy(suggestedReviewers, req.GetSuggestedReviewers())
+		copy(suggestedReviewers[originalNumSuggestedReviewers:], additionalReviewers)
```

Comment:
nit: You can replace it with
```suggestion
		suggestedReviewers := append(originalNumSuggestedReviewers, additionalReviewers...)
```